pentagon/component/gcp/__init__.py

```diff
@@ -0,0 +1,132 @@
+import os
+import json
+import sys
+
+from pentagon.component import ComponentBase
+from apiclient.discovery import build
+
+
+class Cluster(ComponentBase):
+
+    @property
+    def _files_directory(self):
+        return sys.modules[self.__module__].__path__[0] + "/files/cluster"
+
+    def get(self, destination):
+        service = build('container', 'v1')
+        cl = service.projects().zones().clusters()
+        self._zone = self._data.get('zone')
+        self._project = self._data.get('project')
+        self._destination = destination
+        cluster_name = self._data.get('name')
+        if cluster_name:
+            cluster = cl.get(projectId=self._project, zone=self._zone, clusterId=cluster_name).execute()
+            Cluster(self.parse_cluster(cluster)).add(cluster_name)
+            self.add_cluster_nodepools(cluster_name, cluster.get('nodePools'))
+        else:
+            clusters = cl.list(projectId=self._project, zone=self._zone).execute()
+
+            for cluster in clusters['clusters']:
+                Cluster(self.parse_cluster(cluster)).add(cluster['name'])
+                self.add_cluster_nodepools(cluster.get('name'), cluster.get('nodePools'), self._data.get('get_default_nodepools'))
+
+    def parse_cluster(self, cluster):
+        """ parses cluster object returned from api call and converts it to a data strucutur for the .add() method """        
+
+        # remove master zone from additional-zones
+        locations = cluster.get('locations')
+        locations.remove(self._zone)
+
+        for np in cluster.get('nodePools'):
+            if np.get('name') == 'default-pool':
+                default_nodepool = np
```

Comment:
`default_nodepool` doesn't appear to be used from this scope. It can be removed as well as the `if` block.

Reply:
It should be used. There is a bug lower down.